exonum-java-binding/core/src/main/java/com/exonum/binding/util/LibraryLoader.java

```diff
@@ -96,5 +148,26 @@ private static String dynamicLibrariesEnvVar() {
     }
   }
 
-  private LibraryLoader() {}
+  private void checkLibraryVersion() {
+    if (!LIBRARY_VERSION_VERIFICATION_ENABLED) {
+      return;
+    }
+    String nativeLibVersion = nativeGetLibraryVersion();
+    if (!nativeLibVersion.equals(expectedLibVersion)) {
+      String message = String.format(
+          "Mismatch between versions of Java library and native '%s' library:%n"
+              + "  Java library version:   %s%n"
+              + "  Native library version: %s%n"
+              + "Check that the version of 'exonum-java-binding-core' matches the version of "
+              + "the installed 'Exonum Java' application.%n"
+              + "See https://exonum.com/doc/version/0.11/get-started/java-binding/#installation",
+          BINDING_LIB_NAME, expectedLibVersion, nativeLibVersion);
+      logger.fatal(message);
+      throw new LinkageError(message);
+    }
+  }
+
+  // todo: Shall we extract it as a Supplier<String>, and System.load() as Consumer<String>
```

Comment:
We can possibly parameterize this class with this couple of functions (hence the constructor):
```java
private static final LibraryLoader INSTANCE =
    new LibraryLoader(System::loadLibrary, LibraryLoader::nativeGetLibraryVersion,
                      JAVA_BINDING_VERSION);
```

Is this complication worth it? Also, it will still have dependencies on the system properties and environment, unless we also abstract them and parameterize :upside_down_face:

Reply:
I'm not sure about such refactoring for tests because it contains only one `equals` statement.

Reply:
:ok_man: , let's keep it simple(r) then.